commands/utils/packagehandlers/pythonpackagehandler.go

```diff
@@ -0,0 +1,85 @@
+package packagehandlers
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+
+	// Package names are case-insensitive with this prefix
+	PythonPackageRegexPrefix = "(?i)"
+
+	// Match all possible operators and versions syntax
+	PythonPackageRegexSuffix = "\\s*(([\\=\\<\\>\\~]=)|([\\>\\<]))\\s*(\\.|\\d)*(\\d|(\\.\\*))(\\,\\s*(([\\=\\<\\>\\~]=)|([\\>\\<])).*\\s*(\\.|\\d)*(\\d|(\\.\\*)))?"
+)
+
+// PythonPackageHandler Handles all the python package mangers as they share behavior
+type PythonPackageHandler struct {
+	PipRequirementsFile string
+	GenericPackageHandler
+}
+
+func (py *PythonPackageHandler) UpdateImpactedPackage(impactedPackage string, fixVersionInfo *FixVersionInfo, extraArgs ...string) error {
+	// Python package mangers are case-sensitive which can cause duplicates
+	impactedPackage = strings.ToLower(impactedPackage)
+	switch fixVersionInfo.PackageType {
+	case coreutils.Poetry:
```

Comment:
Any reason why we shouldn't create separate package holders for Poetry and Pip?

Reply:
There are shared parameters to all python package handlers , including regex and the need to lower-case
because they don't have any other complex logic i decided to put them together for now.
we can definitely separate them.

Reply:
👍